File: frontend-api/src/main/java/no/nav/melosys/tjenester/gui/graphql/mapping/StrukturertAdresseTilDtoKonverter.java
package no.nav.melosys.tjenester.gui.graphql.mapping;

import no.nav.melosys.domain.FellesKodeverk;
import no.nav.melosys.domain.adresse.StrukturertAdresse;
import no.nav.melosys.service.kodeverk.KodeverkService;
import no.nav.melosys.tjenester.gui.graphql.dto.StrukturertAdresseformatDto;

public final class StrukturertAdresseTilDtoKonverter {
    private StrukturertAdresseTilDtoKonverter() {
    }

    public static StrukturertAdresseformatDto tilDto(StrukturertAdresse strukturertAdresse,
                                                     KodeverkService kodeverkService) {
        if (strukturertAdresse == null) {
            return null;
        }
        return new StrukturertAdresseformatDto(
            strukturertAdresse.getTilleggsnavn(),
            strukturertAdresse.getGatenavn(),
            strukturertAdresse.getHusnummerEtasjeLeilighet(),
            strukturertAdresse.getPostboks(),
            strukturertAdresse.getPostnummer(),
            strukturertAdresse.getPoststed(),
            strukturertAdresse.getRegion(),
            kodeverkService.dekod(FellesKodeverk.LANDKODER_ISO2, strukturertAdresse.getLandkode())
        );
    }
}

File: frontend-api/src/main/kotlin/no/nav/melosys/tjenester/gui/graphql/mapping/StrukturertAdresseTilDtoKonverter.kt
package no.nav.melosys.tjenester.gui.graphql.mapping

import no.nav.melosys.domain.FellesKodeverk
import no.nav.melosys.domain.adresse.StrukturertAdresse
import no.nav.melosys.service.kodeverk.KodeverkService
import no.nav.melosys.tjenester.gui.graphql.dto.StrukturertAdresseformatDto


object StrukturertAdresseTilDtoKonverter {
    fun tilDto(
        strukturertAdresse: StrukturertAdresse?,
        kodeverkService: KodeverkService
    ): StrukturertAdresseformatDto? {
        return if (strukturertAdresse == null) {
            null
        } else StrukturertAdresseformatDto(
            strukturertAdresse.tilleggsnavn,
            strukturertAdresse.gatenavn,
            strukturertAdresse.husnummerEtasjeLeilighet,
            mapPostboks(strukturertAdresse.postboks),
            strukturertAdresse.postnummer,
            strukturertAdresse.poststed,
            strukturertAdresse.region,
            kodeverkService.dekod(FellesKodeverk.LANDKODER_ISO2, strukturertAdresse.landkode)
        )
    }

    private fun mapPostboks(postBoks: String?): String? {
        return if (postBoks != null) "Postboks $postBoks" else null
    }
}

